lib/Posts/LinkTranslation.php

```diff
@@ -160,34 +162,46 @@ public function get_possible_links( WP_Post $post, string $post_lang, int $page,
 			$search_filter = $wpdb->prepare( "AND P.post_title LIKE %s", '%' . $wpdb->esc_like( $search ) . '%' );
 		}
 
+		$instr = [];
+		foreach ( $languages as $lang ) {
+			$instr[] = "INSTR(locale_info, '{$lang}') != 0 as {$lang}";
+		}
+		$instr = implode( ', ', $instr );
+
 		// FIXME: if there are any posts with multiple ubb_sources (bug), it can cause the post to show up in the list multiple times.
 		$possible_sources = $wpdb->get_results(
 			$wpdb->prepare(
-				"SELECT SQL_CALC_FOUND_ROWS
-				MIN(A.post_id) as post_id,
-				GROUP_CONCAT( CONCAT(A.post_id, 0x1F, locale, 0x1F, A.post_title) SEPARATOR 0x1D ) as group_info
+				"SELECT SQL_CALC_FOUND_ROWS post_id, group_info
 				FROM (
-					SELECT PT.post_id, P.post_title, locale, IFNULL(meta_value, PT.post_id) AS source
-					FROM {$translations_table} AS PT
-					LEFT JOIN {$wpdb->postmeta} AS PM ON (PT.post_id = PM.post_id AND meta_key = 'ubb_source')
-					INNER JOIN {$wpdb->posts} as P ON (PT.post_id = P.ID)
-					WHERE post_type = %s AND post_status NOT IN ('revision','auto-draft')
-					AND PT.locale IN ('{$allowed_languages_str}')
-					{$search_filter}
-				) AS A
-				WHERE locale != %s
-				AND source NOT IN (
-					SELECT IFNULL(meta_value, PT.post_id) AS source
-					FROM {$translations_table} AS PT
-					LEFT JOIN {$wpdb->postmeta} AS PM ON (PT.post_id = PM.post_id AND meta_key = 'ubb_source')
-					WHERE locale = %s
-				) GROUP BY source
-				ORDER BY group_info
+					SELECT
+						post_id,
+						source,
+						locale_info,
+						{$instr},
+						group_info
+					FROM (
+						SELECT
```

Comment:
Don't know if this logic is the best approach. The user won't see posts that have the language of the current post. It will feel like "somethings are missing". We should consider just having a simple list of all sources and displaying warnings when the post has a translation for that language.

Reply:
It doesn't make sense for the user to see posts that have the language of the current one. We don't allow for the user to link to those because then the translation group would have two posts for the same language. 

Otherwise, we would be making it an override action, substituting the existing post of that language in that translation group for the one being edited. I believe our early versions of link/unlink had a behaviour like this and we decided agaisnt it.

Reply:
Just because we decided against then, doesn't mean we shouldn't revisit the decision now or in the future. We can keep the current approach and see what the performance is. If it lacks in any way we need to find a better solution. Wither improve the query more or consider an alternate UX.